app/models.js

```diff
@@ -31,6 +35,25 @@ class App extends Model {
       access_level,
     }).create();
   }
+
+  grant_user_access(user, access_level = 'readonly', is_admin = false) {
+    if (!['readonly', 'readwrite'].includes(access_level)) {
+      throw new Error(`Invalid access_level "${access_level}"`);
+    }
+
+    let user_id = user;
+
+    if (typeof user === 'function' && user.prototype.constructor === User) {
+      user_id = user.id;
```

Comment:
Does this work? I don't think the `User` or its base class `Model` have an `id` property (I may be wrong!). You may need to change this to `user.pk` or `user.auth0_id`.

Reply:
Hmmm. Honestly, I copied this from `grant_bucket_access` and didn't overthink it. The only place this gets called sends in a plain `user_id` so this will never get called. I'll strip it out.